### PR TITLE
go.mod: use the latest parser to fix 'select timestampliteral(rand())'

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

```
mysql> select timestampliteral(rand());                                                                                                                                                               ERROR 2013 (HY000): Lost connection to MySQL server during query

mysql> select timestampliteral('2019-10-29 16:40:22');
+-----------------------------------------+
| timestampliteral('2019-10-29 16:40:22') |
+-----------------------------------------+
| 2019-10-29 16:40:22                     |
+-----------------------------------------+
1 row in set (0.00 sec)
```

After the fix:

```
mysql> select timestampliteral(rand());
ERROR 1305 (42000): FUNCTION test.db_9_85b2e2.timestampliteral does not exist
mysql> select timestampliteral('2019-10-29 16:40:22');
ERROR 1305 (42000): FUNCTION test.db_9_85b2e2.timestampliteral does not exist
```

### What is changed and how it works?

Use the latest parser, it renames `timestampliteral` to something else.

### Check List 

Tests 

 - Unit test

Release note

 - Write release note for bug-fix or new feature.

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @tangenta @SunRunAway

**Comment by tiancaiamao** [comment]
PTAL @SunRunAway @tangenta

File: go.mod [hunk 1]
```diff
@@ -42,10 +43,10 @@ require (
 	github.com/pingcap/goleveldb v0.0.0-20171020122428-b9ff6c35079e
 	github.com/pingcap/kvproto v0.0.0-20190910074005-0e61b6f435c1
 	github.com/pingcap/log v0.0.0-20191012051959-b742a5d432e9
-	github.com/pingcap/parser v0.0.0-20191025082927-f8adf1670b97
+	github.com/pingcap/parser v0.0.0-20191030075336-d8fa9d7074cb
 	github.com/pingcap/pd v1.1.0-beta.0.20190923032047-5c648dc365e0
 	github.com/pingcap/tidb-tools v2.1.3-0.20190321065848-1e8b48f5c168+incompatible
-	github.com/pingcap/tipb v0.0.0-20191022094114-a2e8c3fa634b
+	github.com/pingcap/tipb v0.0.0-20191030045153-07a0962bbc64
```
Comment (zz-jason) [inline go.mod]: Is it expected to update `tipb` version in this PR?

Comment (tiancaiamao) [reply go.mod]: I've reset this change...

**Comment by tiancaiamao** [comment]
/run-all-tests

PTAL @SunRunAway @tangenta @zz-jason

**Comment by cyliu0** [comment]
@tiancaiamao Do we need to cherry pick this to release branches?

**Pull request merged**